Implement lookup for full Vault paths and secrets (#30)
inspec/inspec-vault: README.md

## What This Plugin Does

> With the inspec-vault plugin enabled, Chef InSpec will contact the Vault server whenever an `input()` DSL call appears in profile control code. If a secret located at the given location, InSpec will use this value. Otherwise it will fall back to other means of resolving the input, such as other plugins, profile metadata or CLI values.

Reviewer: It looks like a grammar fix is needed here: `If a secret located` --> `If a secret is located`

> With the inspec-vault plugin enabled, Chef InSpec will contact the Vault server whenever an `input()` DSL call appears in profile control code. If a secret located at the given location, InSpec will use this value. Otherwise it will fall back to other means of resolving the input, such as other plugins, profile metadata or CLI values.

Reviewer: By default, this plugin will take precedence over any other input source. But depending on the user's setup; they might have higher precedence input values that take precedence over this plugin.

Instead of trying to fully document precedence in this README, it might be helpful to add this sentence to the end of the paragraph:

> Otherwise it will fall back to other means of resolving the input, such as other plugins, profile metadata or CLI values. See https://www.inspec.io/docs/reference/inputs/ for an explanation on input precedence.

> With the inspec-vault plugin enabled, Chef InSpec will contact the Vault server whenever an `input()` DSL call appears in profile control code. If a secret is located at the given location, InSpec will use this value. Otherwise it will fall back to other means of resolving the input, such as other plugins, profile metadata or CLI values. See https://www.inspec.io/docs/reference/inputs/ for an explanation on input precedence.

Reviewer 2: ```suggestion
With the inspec-vault plugin enabled, whenever an `input()` DSL call appears in profile control code, Chef InSpec contacts the Vault server. If the secret is located in Vault, the InSpec uses this value. Otherwise, it will searches for other sources to resolve the input, such as other plugins, profile metadata or CLI values, as described in the Chef InSpec [input precedence](https://www.inspec.io/docs/reference/inputs/) documentation. 
```

> Chef InSpec will determine a secret lookup path and access Vault, returning the value it found.

Reviewer 2: ```suggestion
```

end
```

> With no other settings, Chef InSpec will look for a Vault secret located at `secret/inspec/my_profile` with a key named `some_input`. The `my_profile` part of the path is the name of your InSpec profile. The `inspec` part comes from the `path_prefix` setting.

Reviewer 2: ```suggestion
 With no other settings, Chef InSpec will look for a Vault secret located at `secret/inspec/my_profile` with a key named `some_input`, where `inspec` is derived from the `path_prefix` setting and `my_profile` is the name of this InSpec profile.
```

### Absolute Path Lookup

> In some cases, you want to access global information and not something related to a specific profile. For this case, qualify the path in absolute syntax with a starting `/`, like this:

Reviewer 2: ```suggestion
To access global information instead of a value related to a specific profile, qualify the path in absolute syntax with a starting `/`. For example:
```

```ruby
describe input("/configuration/webserver/password")
 it { should cmp "some_expected_value" }
end
```

> This will result in access to `secret/configuration/webserver` and the return of value of the `password` key within.

Reviewer 2: ```suggestion
Chef InSpec searches the `secret/configuration/webserver` Vault and returns the value of the `password` key.
```

I'm not sure if this is correct.

Author: Vault is the software, I just learned yesterday that this would be a "document". So that would probably be the right wording.

Author: Added it in my most recent commit
